config/merge/merge_test.go

```diff
@@ -293,6 +293,88 @@ func TestMerge(t *testing.T) {
 			}},
 		},
 
+		// struct pointers
+		// we're not supposed to have any, but some ended up in the
+		// Clevis and Luks structs in spec 3.2.0
+		// https://github.com/coreos/ignition/issues/1132
+		{
+			in1: types.Config{
+				Storage: types.Storage{
+					Luks: []types.Luks{
+						// nested struct pointers, one override
+						{
+							Clevis: &types.Clevis{
+								Custom: &types.Custom{
+									Config: "cfg",
+									Pin:    "pin",
+								},
+							},
+							Device: util.StrToPtr("/dev/foo"),
+							Name:   "bar",
+						},
+					},
+				},
+			},
+			in2: types.Config{
+				Storage: types.Storage{
+					Luks: []types.Luks{
+						// nested struct pointers
+						{
+							Clevis: &types.Clevis{
+								Threshold: util.IntToPtr(2),
```

Comment:
Hmm, a custom config with a threshold is technically an invalid config, but I guess it's good to test that we merge it in right so we can detect its invalidity correctly.

Maybe worth also adding another more realistic test, where e.g. the parent config just uses `tpm2: true`, and the child config sets it to `false` and adds a custom config?

Reply:
The other tests in this file don't use valid configs either.  We're just trying to verify the various merge cases.  You're right that we should be testing an explicit field override, though; added.